lib/models/conversation_item.dart

```diff
@@ -17,7 +17,7 @@ class ConversationItem {
   // Conversation items with the same conversationId are considered equivalent
```

Comment:
I remember why I did this now: if two `ConversationItem` objects are created that refer to the same `Conversation`, the default equivalence relation (identity) will declare them to be different objects but we want them to be considered equivalent (as there is only ever one conversation with a particular conversation id, and thus there should only ever be one `ConversationItem` that refers to the `Conversation`.

Anyway sorry for the bug!